packages/react-storage/src/components/StorageBrowser/actions/handlers/utils.ts

```diff
@@ -50,3 +52,43 @@ export const constructBucket = ({
   bucketName: string;
   region: string;
 } => ({ bucketName, region });
+
+export const parseLocationAccess = (location: LocationAccess): LocationData => {
+  const { permission, scope, type } = location;
+  if (!scope.startsWith('s3://')) {
+    throw new Error(`Invalid scope: ${scope}`);
+  }
+
+  const id = crypto.randomUUID();
+
+  // remove default path
+  const sanitizedScope = scope.slice(5);
+  let bucket, prefix;
+
+  switch (type) {
+    case 'BUCKET': {
+      // { scope: 's3://bucket/*', type: 'BUCKET', },
+      bucket = sanitizedScope.slice(0, -2);
+      prefix = '';
+      break;
+    }
+    case 'PREFIX': {
+      // { scope: 's3://bucket/path/*', type: 'PREFIX', },
+      bucket = sanitizedScope.slice(0, sanitizedScope.indexOf('/'));
+      prefix = `${sanitizedScope.slice(bucket.length + 1, -1)}`;
+
+      break;
+    }
+    case 'OBJECT': {
+      // { scope: 's3://bucket/path/to/object', type: 'OBJECT', },
+      bucket = sanitizedScope.slice(0, sanitizedScope.indexOf('/'));
+      prefix = sanitizedScope.slice(bucket.length + 1);
+      break;
+    }
+    default: {
+      throw new Error(`Invalid location type: ${type}`);
+    }
+  }
+
+  return { bucket, id, permission, prefix, type };
```

Comment:
Nit: Could return within the cases to avoid having to break or use `let`

Reply:
That's what i normally what i would do here but am trying to actually use `let` instead of forcing `const` everywhere 😓

Reply:
Hmm I'm curious what the rationale for that is. `let` is a perfectly acceptable pattern when it is necessary but why wouldn't we discourage it when it isn't necessary but _does_ introduce potential sharp edges?

Reply:
Not quite sure i see sharp edges here, would be more concerned if the assignment was an array or object